lib/MusicBrainz/Server/Data/Artist.pm

```diff
@@ -465,12 +466,27 @@ sub _hash_to_row
 }
 
 sub load_related_info {
-    my ($self, @artists) = @_;
+    my ($self, $artists_ref, $user_lang) = @_;
 
     my $c = $self->c;
+    my @artists = @{$artists_ref};
     $c->model('ArtistType')->load(@artists);
     $c->model('Gender')->load(@artists);
     $c->model('Area')->load(@artists);
+
+    # Load and save aliases so Controller::Artist::show can use them later.
+    my $artist_aliases = $c->model('Artist')->alias->find_by_entity_ids(
+        map { $_->id } @artists,
+    );
+    for my $artist (@artists) {
+        my @aliases = @{ $artist_aliases->{$artist->id} };
+        $c->model('Artist')->alias_type->load(@aliases);
+        $artist->{aliases} = \@aliases;
```

Comment:
Moose accessors can be called like setters:
```suggestion
        $artist->aliases(\@aliases);
```
Setting the property directly doesn't perform any validation or trigger other Moosey configuration on the attribute.